storage/innobase/buf/buf0lru.cc

```diff
@@ -743,6 +743,9 @@ buf_flush_dirty_pages(
 		err = buf_flush_or_remove_pages(
 			buf_pool, id, observer, flush, trx);
 
+	        ut_ad(err == DB_INTERRUPTED || err == DB_FAIL
```

Comment:
I have moved it so the check for buf_pool_get_dirty_pages_count was made while a mutex LRU_list_mutex is locked. Otherwise it could fail due to multiple encryption threads. One of the threads could start some work on the space and there could be some dirty pages in the buffer when this assert was reached.